server/finders/BookFinder.js

```diff
@@ -179,35 +188,166 @@ class BookFinder {
+          this.positions[candidate] = position
+        }
+        this.priorities[cleanTitle] = 1
+      }
+    }
+
+    get size() {
+      return this.candidates.size
+    }
+
+    getCandidates() {
+      var candidates = [...this.candidates]
+      candidates.sort((a, b) => {
+        // Candidates that include the author are likely low quality
+        const includesAuthorDiff = !b.includes(this.cleanAuthor) - !a.includes(this.cleanAuthor)
+        if (includesAuthorDiff) return includesAuthorDiff
+        // Candidates that include only digits are also likely low quality
+        const onlyDigits = /^\d+$/
+        const includesOnlyDigitsDiff = !onlyDigits.test(b) - !onlyDigits.test(a)
+        if (includesOnlyDigitsDiff) return includesOnlyDigitsDiff
+        // transformed candidates receive higher priority
+        const priorityDiff = this.priorities[a] - this.priorities[b]
+        if (priorityDiff) return priorityDiff
+        // if same priorirty, prefer candidates that are closer to the beginning (e.g. titles before subtitles)
+        const positionDiff = this.positions[a] - this.positions[b]
+        if (positionDiff) return positionDiff
+        // Start with longer candidaets, as they are likely more specific
+        const lengthDiff = b.length - a.length
+        if (lengthDiff) return lengthDiff
+        return b.localeCompare(a)
+      })
+      Logger.debug(`[${this.constructor.name}] Found ${candidates.length} fuzzy title candidates`)
+      Logger.debug(candidates)
+      return candidates
+    }
+
+    delete(title) {
+      return this.candidates.delete(title)
+    }
+  }
+
+  static AuthorCandidates = class {
+    constructor(bookFinder, cleanAuthor) {
+      this.bookFinder = bookFinder
+      this.candidates = new Set()
+      this.cleanAuthor = cleanAuthor
+      if (cleanAuthor) this.candidates.add(cleanAuthor)
+    }
+
+    validateAuthor(name, region = '', maxLevenshtein = 2) {
+      return this.bookFinder.audnexus.authorASINsRequest(name, region).then((asins) => {
+        for (const [i, asin] of asins.entries()) {
+          if (i > 10) break
+          let cleanName = this.bookFinder.cleanAuthorForCompares(asin.name)
+          if (!cleanName) continue
+          if (cleanName.includes(name)) return name
+          if (name.includes(cleanName)) return cleanName
+          if (levenshteinDistance(cleanName, name) <= maxLevenshtein) return cleanName
+        }
+        return ''
+      })
+    }
+
+    add(author) {
+      const authorTransformers = []
+
+      // Main variant
+      const cleanAuthor = this.bookFinder.cleanAuthorForCompares(author).trim()
+      if (!cleanAuthor) return false
+      this.candidates.add(cleanAuthor)
+
+      let candidate = cleanAuthor
+
+      for (const transformer of authorTransformers) {
+        candidate = candidate.replace(transformer[0], transformer[1]).trim()
+        if (candidate) {
+          this.candidates.add(candidate)
+        }
+      }
```

Comment:
This is unused, were you planning on adding to this?

Reply:
This was meant for future additions, but I wasn't planning on having any just right now. I removed it (will add it back if I need it).